internal/core/command/get_test.go

```diff
@@ -108,9 +118,78 @@ func TestNewGetCommandNoCorrelationIDInContext(t *testing.T) {
 
 func TestNewGetCommandInvalidBaseUrl(t *testing.T) {
 	device := testDevice
+	req := newRequestForGetWithContentType(clients.ContentTypeCBOR)
 	device.Service.Addressable.Address = "!@#$"
-	_, err := NewGetCommand(device, testCommand, "", context.Background(), nil, logger.NewMockClient())
+	_, err := NewGetCommand(device, testCommand, context.Background(), nil, logger.NewMockClient(), req)
 	if err == nil {
 		t.Errorf("The invalid URL error was not properly propagated to the caller")
 	}
 }
+
+func TestNewGetCommandContentType(t *testing.T) {
+	tests := []struct {
+		name               string
+		expectedHeaderType string
+		req                *http.Request
+	}{
+		{
+			"cbor content type",
+			clients.ContentTypeCBOR,
+			newRequestForGetWithContentType(clients.ContentTypeCBOR),
+		},
+		{
+			"json content type",
+			clients.ContentTypeJSON,
+			newRequestForGetWithContentType(clients.ContentTypeJSON),
+		},
+		{
+			"no content type was provided",
+			"",
+			newRequestForGetWithContentType(""),
+		},
+		{
+			"user-agent header should not be propagated",
+			"",
+			newRequestForGetWithUnexpectedUserAgentHeader("User-Agent"),
+		},
+		{
+			"content-md5 header should not be propagated",
+			"",
+			newRequestForGetWithUnexpectedContentMD5Header("Content-MD5"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var loggerMock = logger.NewMockClient()
+			ctx := context.Background()
+			getCommand, _ := NewGetCommand(
+				testDevice,
+				testCommand,
+				ctx,
+				nil,
+				loggerMock,
+				tt.req)
+			actualContentType := getCommand.(serviceCommand).Request.Header.Get(clients.ContentType)
+			require.Equal(t, tt.expectedHeaderType, actualContentType)
+		})
+	}
+}
+
+func newRequestForGetWithContentType(contentType string) *http.Request {
+	req, _ := http.NewRequest(http.MethodGet, "/", strings.NewReader("Test body"))
+	req.Header.Set(clients.ContentType, contentType)
+	return req
+}
+
+func newRequestForGetWithUnexpectedUserAgentHeader(headerType string) *http.Request {
+	req, _ := http.NewRequest(http.MethodGet, "/", strings.NewReader("Test body"))
+	req.Header.Set("User-Agent", headerType)
+	return req
+}
+
+func newRequestForGetWithUnexpectedContentMD5Header(headerType string) *http.Request {
+	req, _ := http.NewRequest(http.MethodGet, "/", strings.NewReader("Test body"))
+	req.Header.Set("Content-MD5", headerType)
+	return req
+}
```

Comment:
Can we consolidate these functions to one general function which accepts the header name and value? Furthermore, I think you can make the tests cover more functional cases by having the general function accept multiple headers that way we can test cases where there are multiple headers. For example, requests with headers we want to propagate, and ones we do not want. The tests as they are only test that cases where only one header present, we have not really verified that the functionality properly handles multiple headers.